### PR TITLE
change web element location finding semantics

### PR DESCRIPTION
**Pull request by jlipps**
before we were returning location in view, because that was what the atom did. but we should be returning location relative to 0, 0 of the document. note that this is a breaking change so we might want to think about when and how to land it. this change would also need to be done in ios-driver if we want them to stay in sync, since the code is duplicated from there.

### CODE REVIEW

File: lib/commands/element.js [hunk 1]
```diff
@@ -45,8 +45,13 @@ commands.getRect = async function (el) {
 commands.getLocation = async function (el) {
   el = el.ELEMENT ? el.ELEMENT : el;
   if (this.isWebContext()) {
-    let atomsElement = await this.useAtomsElement(el);
-    return await this.executeAtom('get_top_left_coordinates', [atomsElement]);
+    const atomsElement = await this.useAtomsElement(el);
+    const script = "return [document.body.scrollLeft, document.body.scrollTop];";
+    const [xOffset, yOffset] = await this.execute(script);
+    let loc = await this.executeAtom('get_top_left_coordinates', [atomsElement]);
```
Comment (mykola-mokhnach) [inline lib/commands/element.js]: can be const as well

File: lib/commands/element.js [hunk 2]
```diff
@@ -45,8 +45,13 @@ commands.getRect = async function (el) {
 commands.getLocation = async function (el) {
   el = el.ELEMENT ? el.ELEMENT : el;
   if (this.isWebContext()) {
-    let atomsElement = await this.useAtomsElement(el);
-    return await this.executeAtom('get_top_left_coordinates', [atomsElement]);
+    const atomsElement = await this.useAtomsElement(el);
+    const script = "return [document.body.scrollLeft, document.body.scrollTop];";
```
Comment (mykola-mokhnach) [inline lib/commands/element.js]: we usually use single quotes %)

File: lib/commands/element.js [hunk 3]
```diff
@@ -45,8 +45,13 @@ commands.getRect = async function (el) {
...
+    const [xOffset, yOffset] = await this.execute(script);
+    let loc = await this.executeAtom('get_top_left_coordinates', [atomsElement]);
+    loc.x += xOffset;
+    loc.y += yOffset;
+    return loc;
```
Comment (mykola-mokhnach) [inline lib/commands/element.js]: probably, we can have a new capability, which controls whether to keep the old approach?

Comment (imurchie) [inline lib/commands/element.js]: I agree. Then later we can break things by changing the default.

Comment (jlipps) [inline lib/commands/element.js]: ok, will hide by a capability

File: lib/commands/element.js (hunk 1 above)
Comment (jlipps) [inline lib/commands/element.js]: aren't i mutating the properties of this object though?

**Comment by jlipps** [comment]
ok pushed some changes and added tests. i also ran a functional test locally to prove the cap worked and returned the right thing. it does.

File: lib/commands/element.js (hunk 1 above)
Comment (mykola-mokhnach) [inline lib/commands/element.js]: _const_ does not make the value immutable. See http://exploringjs.com/es6/ch_variables.html#ch_variables, p. 9.3.1 for the reference ;)

**Comment by mykola-mokhnach** [comment]
shall we add a description of the new capability to README?

File: lib/commands/element.js (hunk 1 above)
Comment (jlipps) [inline lib/commands/element.js]: yes i know about this behavior, but it seems odd to mark something as `const` and then change something inside it. i consider it a weakness of `const` and is one reason i use `const` more as code documentation meaning "i don't want to change any part of this" than anything else.

Comment (mykola-mokhnach) [inline lib/commands/element.js]: fair point. Missing C++ const notation? :)

Comment (jlipps) [inline lib/commands/element.js]: yeah a little bit :-) just seems weird to have a `const` which is so weak.

**Comment by pluie-cloud** [comment]
ERROR: Cannot find module 'eslint-config-appium' when i run ./Scripts/bootstrap.sh –d

**Comment by imurchie** [comment]
@pluie-cloud Why are you making that comment on this pull request? 

Also, make sure you are using the `-d` flag.

**Comment by mykola-mokhnach** [comment]
@imurchie shall we close this PR?

**Comment by imurchie** [comment]
I'm not sure what @jlipps intentions are with this.

**Comment by jlipps** [comment]
my intentions are that it be merged! let me try to resolve conflicts

**Comment by jlipps** [comment]
ok, conflicts fixed. i think this is mergeable now.

**Comment by dpgraham** [comment]
Lgtm. @imurchie are you okay with me merging this?

**Comment by imurchie** [comment]
I'd rather wait for at least one run of the tests. Otherwise what's the point of CI?

**Pull request merged**